app/src/main/java/ru/practicum/android/diploma/search/presentation/ui/SearchFragment.kt

```diff
@@ -97,78 +112,98 @@ class SearchFragment : BindingFragment<FragmentSearchBinding>() {
                 }
             }
         })
-        setupDefaultUI()
+        if (doSearchAgain) {
+            setupDefaultUI()
+        }
         setupTextWatcher()
         setupRecyclerView()
     }
 
     private fun debounceSearch(p0: String) {
+        refreshList = true
         timerJob?.cancel()
         timerJob = viewLifecycleOwner.lifecycleScope.launch {
             delay(SEARCH_DEBOUNCE_DELAY_MILLIS)
-            viewModel.searchVacancies(p0)
+            clearUI()
+            viewModel.newSearchVacancies(p0)
+            binding.recyclerView.scrollToPosition(0)
+            binding.progressBarLoader.visibility = View.VISIBLE
+            binding.searchEditText.isFocusable = false
+            binding.searchEditText.isFocusableInTouchMode = true
+            val imm =
+                requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+            imm.hideSoftInputFromWindow(view?.windowToken, 0)
         }
     }
 
     private fun setupDefaultUI() {
         with(binding) {
             searchPlaceholder.setImageResource(R.drawable.placeholder_search)
-            if (searchEditText.text.isNullOrEmpty()) {
+            if (searchEditText.text.isEmpty()) {
                 clearUI()
             }
         }
-        println("Test")
     }
 
     private fun observeViewModel() {
         viewModel.searchVacancyResult.observe(viewLifecycleOwner) { state ->
-            binding.iconSearch.setImageResource(R.drawable.ic_search)
-            when (state) {
+            when (state.result) {
                 is SearchVacancyResult.Error -> updateUI(SearchUIState.CONNECTION_ERROR)
                 SearchVacancyResult.EmptyResult -> updateUI(SearchUIState.EMPTY_SEARCH)
                 SearchVacancyResult.NoInternet -> updateUI(SearchUIState.NO_INTERNET)
                 is SearchVacancyResult.Success -> {
-                    viewModel.maxPagesSet(state.response.pages)
+                    viewModel.maxPagesSet(state.result.response.pages)
                     viewModel.isNextPageLoading = true
-                    showVacancy(state.response.vacancies)
+                    showVacancy(state.result.response.vacancies)
                 }
-                SearchVacancyResult.Loading -> {}
-                is SearchVacancyResult.StartScreen -> setFilterIcon(state.isFiltersOn)
+                SearchVacancyResult.Loading -> updateUI(SearchUIState.LOADING)
+                else -> {}
```

Comment:
🍏 Вместо создания объекта пустой лямбды можно сразу указат возвращаемое значение
```suggestion
                else -> Unit
```